parsers/BB.py

```diff
@@ -0,0 +1,181 @@
+from datetime import datetime, timedelta
+from logging import Logger, getLogger
+from typing import Any
+from zoneinfo import ZoneInfo
+
+import pandas as pd
+
+# The request library is used to fetch content through HTTP
+import requests
+from requests import Session
+from requests.auth import AuthBase
+
+from electricitymap.contrib.lib.models.event_lists import (
+    ProductionBreakdownList,
+    TotalConsumptionList,
+)
+from electricitymap.contrib.lib.models.events import ProductionMix, StorageMix
+from electricitymap.contrib.lib.types import ZoneKey
+from parsers.lib.exceptions import ParserException
+
+base_url = "https://dataservices.acelerex.com/api"
+
+
+class JwtPublicAuth(AuthBase):
+    def __init__(self):
+        self.token = None
+
+    def __call__(self, r):
+        if self.token is None:
+            self.token = self.get_token()
+        r.headers["Authorization"] = f"Bearer {self.token}"
+        return r
+
+    def get_token(self) -> str:
+        url = base_url + "/authentication"
+        response = requests.post(
+            url, json={"dashboardId": "barbados", "strategy": "jwt-public"}
+        )
+        return response.json()["accessToken"]
```

Comment:
The token is issued for one hour. Will a single session ever run for longer than that? Is historical data fetched in a single session?

Wondering if implementing token renewal is needed.

Reply:
Should not be needed as far as I know, but it also wouldn't hurt to have it in there if you want to do it.